src_c/rwobject.c

```diff
@@ -694,21 +688,20 @@ _rwops_from_pystr(PyObject *obj)
     rw = SDL_RWFromFile(encoded, "rb");
 
     if (rw) {
-        /* adding the extension to the hidden data for RWops from files */
-        /* this is necessary to support loading functions that rely on
-         * file extensions in a convenient way. File-like objects use this
-         * field for a helper object. */
-        char *extension = NULL;
-        char *ext = strrchr(encoded, '.');
-        if (ext && strlen(ext) > 1) {
-            ext++;
-            extension = malloc(strlen(ext) + 1);
-            if (!extension) {
-                return (SDL_RWops *)PyErr_NoMemory();
+        /* If a valid extptr has been passed, populate it with a dynamically
+         * allocated field for the file extension. */
+        if (extptr) {
+            char *ext = strrchr(encoded, '.');
+            if (ext && strlen(ext) > 1) {
+                ext++;
+                *extptr = malloc(strlen(ext) + 1);
+                if (!(*extptr)) {
+                    return (SDL_RWops *)PyErr_NoMemory();
```

Comment:
This return does not seem to have a `Py_DECREF(oencoded);`, but there is one in the happy path.  I suspect this is a leak-on-error case

Reply:
Also, shouldn't we clean up `rw` as well in case of error?

Reply:
Good spot! Pushed a commit to resolve this.